tests/cupyx_tests/scipy_tests/fftpack_tests/test_fftpack.py

```diff
@@ -88,6 +88,38 @@ def test_fft2_overwrite(self, xp, scp, dtype):
         return scp.fftpack.fft2(x, shape=self.s, axes=self.axes,
                                 overwrite_x=True)
 
+    @testing.for_complex_dtypes()
+    @testing.numpy_cupy_allclose(rtol=1e-4, atol=1e-7, accept_error=ValueError,
+                                 contiguous_check=False, scipy_name='scp')
+    def test_fft2_plan(self, xp, scp, dtype):
+        x = testing.shaped_random(self.shape, xp, dtype)
+        if isinstance(scp, cupyx.scipy):
+            import cupy.fft.config as config
+            config.enable_nd_planning = False  # use explicit plan
+            plan = scp.fftpack.get_fft_plan(x)
+            out = scp.fftpack.fft2(x, shape=self.s, axes=self.axes, plan=plan)
```

Comment:
This test fails when `self.s` is not `None`.

Reply:
Sorry, can you be clearer? Does it have to do with `ndim == 4` only? And by failure you meant the results don't match, or there's an exception? Thanks!

Reply:
does it fail only with explicit plan? I'm seeing something strange. Looking into it...

Reply:
@asi1024 I need more information from you for reproducing the failure. In my tests I could only tell that when `self.s` is added, `get_fft_plan` needs to know the "cooked" shape. I will add this support to the upcoming commit.

However, I don't think my finding is relevant to the error you saw. I suspect it is caused by this test case:
https://github.com/cupy/cupy/blob/a7803c638ef9221c1eed030fa240af4f4b4c9441/tests/cupyx_tests/scipy_tests/fftpack_tests/test_fftpack.py#L67

which asks for transforming 4 axes and so is not supported by the current cuFFT plan_nd implementation *by design*. The error message would be `CUFFT can only transform along 1, 2 or 3 axes, but 4 axes were specified.` Am I right? If so, I'll just mark this test case so that when an explicit plan is used we skip this one.

Reply:
The test failure is caused by incompatibility between `scipy` and `cupyx.scipy`. (Only cupyx raises the error which you wrote.) How about returning `None` instead of raising `ValueError` in `get_fft_plan`?

Reply:
Dear @asi1024, thanks a lot for your reply and patience! I finally figure out how to do the tests myself using `unittest`. I can confirm several observations made along this conversion but I also find more, see the summary below. A bug-free commit ef46518 is just pushed.

Here's a list of issues I resolved:
- `self.s` needs to be passed to the FFT functions in `scipy.fftpack` or their CuPy counterparts.
- when the above is done, the input shape could be changed internally (and so is the output shape) before passing to the underlying FFT library. The cuFFT plan needs to be informed of this.
- need an assignment when `overwrite_x=True`, see #1988.
- `axes` could be `None` in `get_fft_plan()`.
- in some cases we cannot generate a cuFFT `PlanNd`, so in such situations we skip the test, see below.

Now, I can see that sometimes returning `None` is preferred over raising an exception. However, when `PlanNd` was implemented (not by me), it was mainly targeting at boosting FFT performance. It is thus expected that people who would call `get_fft_plan()` are somewhat more advanced users, and they would like to really get a plan to speed up the FFT part in their applications. 

With this background, I would tend to object to this design pattern for `get_fft_plan()` for three reasons:
1. Returning `None` will kill the expected speedup silently, and users would not know why and would be puzzled for a while until they check the returned plan and find out it's `None`.
2. Even if they do the check, there'd be no instruction as for why `None` was returned. Raising an exception tells them the reason so that they can make necessary changes if possible.
3. We don't want to change significantly the implantation just to get some tests passed, especially when those tests are explicitly excluded by original design. We only do this if there's a compelling argument.

Therefore, I conclude that our lives would be much simpler if the offending tests are simply skipped. Let me know what you think. If you are convinced please re-run the test for me. Thank you 🙏

Reply:
I am also +1 on not silently returning `None` instead of a plan. I agree that we should just skip test cases where n-dimensional planning is not possible. 

If we want to be able to allow preplanning for these other, more general cases, it would be possible to extend `get_fft_plan` in the future so that it could return, for example, a list of 1d plans for cases where n-dimensional planning is not possible.  The fft functions would then need to also be modified to handle lists of plans in addition to a single plan.